src/com/google/enterprise/adaptor/database/ValidatedUri.java

```diff
@@ -67,42 +66,29 @@ private static String fixupUri(String uriString) {
     String[] parts = uriString.split("\\?", 2);
     StringBuilder builder = new StringBuilder();
     percentEncode(builder, parts[0].replace('\\', '/'));
-    if (parts.length == 1) {
-      return builder.toString();
-    }
-    builder.append('?');
-    StringTokenizer tokenizer = new StringTokenizer(parts[1], "&=#", true);
-    while (tokenizer.hasMoreTokens()) {
-      String token = tokenizer.nextToken();
-      switch (token) {
-        case "&":
-        case "=":
-        case "#":
-          builder.append(token);
-          break;
-        default:
-          percentEncode(builder, token.replace(' ', '+'));
-          break;
-      }
+    if (parts.length == 2) {
+      builder.append('?');
+      percentEncode(builder, parts[1]);
     }
     return builder.toString();
   }
 
-  private static final String PERCENT = "+%-./0123456789:%%%%%"
-      + "%ABCDEFGHIJKLMNOPQRSTUVWXYZ[%]%_"
+  private static final String PERCENT =
```

Comment:
Add a comment about which characters *are* encoded, or even build this string from such a string.